### PR TITLE
User Friendly Merge File Selection

### LINKED ISSUE #1174: Merge File Selection
# Deletion of Files when using Merge

In my opinion, the process of selecting files for the merge operation is a bit clunky. An example to understand what I mean:

1. Select one or more files via the "Select File" button provided.
2. Repeat this process --> Files that have been added before are now deleted.

## Proposed Changes:

1. **Do not delete the files that have been added before**: For instance, if you forgot one file, you shouldn't have to select all of them again along with the extra file to merge them.
   
2. **Handling duplicates**: Duplicate files can already be added using drag and drop. Implementing the changes would mean that this can also be done with the file selector. In my opinion, it should be possible to add a file more than once, but maybe the UI should give a little warning in case it was a mistake.

If you approve this, I would like to start working on implementing these changes.

### PR DESCRIPTION
**Pull request by t71rs**
# User Friendly Merge File Selction

If you had already added files to Merge them, but forgot one or more files and you went ahead and pressed the "Select Files" button, all previously selected Files were deleted.
I fixed that and made it possible to add Files "File by File". Additionally if someone should add the same File twice, a duplicate warning is now displayed, which is removed, if the duplicates are removed. Merging duplicates is possibel though if someone should want to do that. To work with duplicates I added a unique id for each added FIle in the merging process. This makes it possible to differ between files that have the same Name.

Closes #1174 

## Checklist:

- [X] I have read the [Contribution Guidelines](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/CONTRIBUTING.md)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings

## Contributor License Agreement

By submitting this pull request, I acknowledge and agree that my contributions will be included in Stirling-PDF and that they can be relicensed in the future under the MPL 2.0 (Mozilla Public License Version 2.0) license.

(This does not change the general open-source nature of Stirling-PDF, simply moving from one license to another license)

### CODE REVIEW

**Comment by Frooodle** [comment]
UI has been updated can you confirm this works with latest updates

**Comment by t71rs** [comment]
Yes I have just tested it locally with the new UI (looks very nice btw :) ) and it still works!

**Review by Frooodle — APPROVED** [review]

File: src/main/resources/static/js/merge.js [hunk 1]
```diff
@@ -2,28 +2,58 @@ let currentSort = {
...
+    const fileNameDiv = document.createElement("div");
+    fileNameDiv.className = "filename";
+    fileNameDiv.textContent = files[i].file.name;
+
+    // Check for duplicates and add a warning if necessary
+    const duplicateFiles = files.filter(file => file.file.name === files[i].file.name);
+    if (duplicateFiles.length > 1) {
+      const warning = document.createElement("span");
+      warning.className = "duplicate-warning";
+      warning.textContent = "(Duplicate)";
```
Comment (Frooodle) [inline src/main/resources/static/js/merge.js:49]: Should be translatable and dynamic

Comment (Frooodle) [reply src/main/resources/static/js/merge.js:49]: will add as todo before next release

Comment (t71rs) [reply src/main/resources/static/js/merge.js:49]: Somehow did not think about that. I can try and do that as well.

**Comment by Frooodle** [comment]
This has broken logic for ALL other form inputs with files, now each forms inputs doesnt wipe the previous causing very bad usage for others

**Comment by Frooodle** [comment]
reverting until it can be fixed! :)

**Comment by t71rs** [comment]
Damn I'm sorry, I will look into it and think of a solution that wont affect anything else since I quite liked this change.

**Pull request merged**